pkg/builder/builder.go

```diff
@@ -110,6 +110,67 @@ func (b *Builder) Build(cacheDir string) error {
 	return nil
 }
 
+func (b *Builder) BuildWithDependency(cacheDir string) error {
+	indexDir := filepath.Join(cacheDir, types.IndexesDir)
+	count, err := fileutil.Count(indexDir)
+	if err != nil {
+		return xerrors.Errorf("count error: %w", err)
+	}
+	bar := pb.StartNew(count)
+	defer log.Println("Build completed")
+	defer bar.Finish()
+
+	var indexes []types.Index
+	if err := fileutil.Walk(indexDir, func(r io.Reader, path string) error {
+		index := &crawler.Index{}
+		if err := json.NewDecoder(r).Decode(index); err != nil {
+			return xerrors.Errorf("failed to decode index: %w", err)
+		}
+		for _, ver := range index.Versions {
+			indexes = append(indexes, types.Index{
+				GroupID:     index.GroupID,
+				ArtifactID:  index.ArtifactID,
+				Version:     ver.Version,
+				SHA1:        ver.SHA1,
+				ArchiveType: index.ArchiveType,
+				Dependency:  ver.Dependency,
+			})
+		}
+		bar.Increment()
+
+		if len(indexes) > 1000 {
+			if err = b.db.InsertIndexesWithDependency(indexes); err != nil {
+				return xerrors.Errorf("failed to insert index to db: %w", err)
+			}
+			indexes = []types.Index{}
+		}
+		return nil
+	}); err != nil {
+		return xerrors.Errorf("walk error: %w", err)
+	}
+
+	// Insert the remaining indexes
+	if err = b.db.InsertIndexesWithDependency(indexes); err != nil {
+		return xerrors.Errorf("failed to insert index to db: %w", err)
+	}
+
+	if err := b.db.VacuumDB(); err != nil {
+		return xerrors.Errorf("fauled to vacuum db: %w", err)
+	}
+
+	// save metadata
+	metaDB := db.Metadata{
+		Version:    db.SchemaVersion,
```

Comment:
Any reason for this? Since the codebase is the same, wouldnt it be cumbersome to track changes to both javadb and javadepdb and change only one?

Reply:
Since we are treating them as separate DBs, independent versioning might be ideal . Assume we make changes to the GAV db we'll end up updating the version of dep DB as well automatically and vice versa